test/testdata/parser/method_def_trailing_comma.rb

```diff
@@ -0,0 +1,10 @@
+# typed: false
+
+def comma_after_pos_args(x, y,)
+end
```

Comment:
Two questions:

1. Do we need tests for splat/optional/optional kwarg/kwsplats?
2. I imagine other parser changes that you're making have tests to verify that we get reasonable parse trees from erroneous output, which is great.  But do we want to put such tests in a separate directory to make it clearer that they are testing error recovery in some way?

Reply:
> Do we need tests for splat/optional/optional kwarg/kwsplats?

Sure, I can add some. I don't think I handled those cases, because I think they're less common, but may as well as long as they don't provide conflicts.

If they don't already work, I'll probably punt on addressing them to a follow up PR.

> But do we want to put such tests in a separate directory to make it clearer that they are testing error recovery in some way?

I don't think there's a ton of value in centralizing them in one place. The fact that tests have `error: unexpected token` assertions in them should suggest that they're testing parse error recovery. I've been putting most of the tests in the `test/testdata/lsp/completion/` folder actually, because most of them in addition to having a parse tree exp file have `completion` assertions.

Again, seems like purely aesthetic, I'm happy to consolidate but I don't see a lot of addition value over just having the test at all. lmk.

Reply:
> Sure, I can add some. I don't think I handled those cases, because I think they're less common, but may as well as long as they don't provide conflicts.

I looked into this, and it looks like it'll be a bit of a combinatorial explosion to write rules for these cases. I'm going to punt on them for this PR.

Reply:
https://github.com/sorbet/sorbet/issues/5120